src/views/NewShipment.vue

```diff
@@ -0,0 +1,1198 @@
+      this.basecost = this.$store.state.NewShipment.basecost;
+    });
+  }
+
+  validate(page: number) {
+    if (page == 2) {
+      if (this.$refs.form1.validate()) {
+        if (this.selectedOptions != null) {
+          for (let i = 0; i < this.selectedOptions.length; i++) {
+            this.Order.options.push({ option: this.selectedOptions[i] });
+            this.OptionsTotal = this.CalculateOptionsCharge(
+              this.selectedOptions[i]
+            );
+          }
+        }
+        this.e1 = page;
+      }
+    } else if (page == 3) {
+      if (this.Order.packages.length > 0) {
+        this.CalculateTotal(0);
+        this.e1 = page;
+      } else {
+        this.$refs.form2.validate();
+      }
+    }
+  }
+
+  changePage(link: string) {
+    this.$router.push({ name: link });
+  }
+
+  addPackage() {
+    console.log("Package details", this.PackageDetails);
+    if (this.$refs.form2.validate()) {
+      this.Order.packages.push({
+        width: this.PackageDetails.width,
+        height: this.PackageDetails.height,
+        length: this.PackageDetails.length,
+        weight: this.PackageDetails.weight,
+        characteristic: this.PackageDetails.characteristics,
+        description: this.PackageDetails.description,
+        cost: this.CalculatePackageTotal(
+          this.PackageDetails.weight,
+          this.PackageDetails.height,
+          this.PackageDetails.width,
+          this.PackageDetails.length,
+          this.PackageDetails.characteristics
+        ),
+      });
+    }
+    this.$refs.form2.reset();
+  }
+  characteristicCharge: any;
+  CalculatePackageTotal(
+    PackWeight: any,
+    PackHeight: any,
+    PackWidth: any,
+    PackLenght: any,
+    PackCharacteristic: any
+  ) {
+    if (PackCharacteristic != null) {
+      this.characteristicCharge = this.characteristics.filter(
+        (c) => c.ch_id == PackCharacteristic
+      )[0].ch_charge;
+    } else {
+      this.characteristicCharge = 0;
+    }
+
+    this.GrossWeight = this.basecost[1].co_value * PackWeight;
+    this.DimensionalWeight = (
+      (this.basecost[1].co_value * PackWidth * PackLenght * PackHeight) /
+      5000
+    ).toFixed(2);
+    if (this.GrossWeight > this.DimensionalWeight) {
+      return this.GrossWeight + this.characteristicCharge;
+    } else {
+      return this.DimensionalWeight + this.characteristicCharge;
+    }
+  }
+
+  CalculateOptionsCharge(i: number) {
+    if (this.selectedOptions != null) {
+      this.OptionsCharge.push(
+        this.options.filter((o) => o.op_id == i)[0].op_charge
+      );
+    } else {
+      this.OptionsCharge = [];
+    }
+  }
+
+  CalculateOptionsTotal() {
+    if (this.OptionsCharge != null) {
+      this.ShipmentSurcharges = this.OptionsCharge.reduce(
+        (total: any, num: any) => total + num,
+        0
+      ).toFixed(2);
+      return this.ShipmentSurcharges;
+    } else {
+      return 1;
+    }
+  }
+
+  CalculatePackagesTotal() {
+    return this.Order.packages
```

Comment:
return this.Order.packages
      .reduce(
        (acc: any, item: { cost: string | number | null }) =>
          acc + parseFloat(item.cost),
        0
      )
      .toFixed(2);

Reply:
No me acepta el tipo string porque ese atributo en Order.packages es number | null.

Reply:
Listo! Hice otras modificaciones pero ya solucionado